php/php.editor/src/org/netbeans/modules/php/editor/PredefinedSymbols.java

```diff
@@ -107,13 +107,57 @@ public final class PredefinedSymbols {
                 "__unserialize", // NOI18N PHP 7.4
             })));
 
+    public static final Set<String> ATTRIBUTE_NAMES = Collections.unmodifiableSet(Attributes.ATTRIBUTE_NAMES);
+    public static final Set<String> ATTRIBUTE_FQ_NAMES = Collections.unmodifiableSet(Attributes.ATTRIBUTE_FQ_NAMES);
+
     public static enum VariableKind {
         STANDARD,
         THIS,
         SELF,
         PARENT
     };
 
+    // https://www.php.net/manual/en/reserved.attributes.php
+    public static enum Attributes {
+        ATTRIBUTE("Attribute"), // NOI18N
+        ALLOW_DYNAMIC_PROPERTIES("AllowDynamicProperties"), // NOI18N
+        OVERRIDE("Override"), // NOI18N
+        RETURN_TYPE_WILL_CHANGE("ReturnTypeWillChange"), // NOI18N
+        SENSITIVE_PARAMETER("SensitiveParameter"), // NOI18N
+        ;
+
+        private final String name;
+        private final String fqName;
+        private final String asAttributeExpression;
+        private static final Set<String> ATTRIBUTE_NAMES = new HashSet<>();
+        private static final Set<String> ATTRIBUTE_FQ_NAMES = new HashSet<>();
+
+        static {
+            for (Attributes attribute : Attributes.values()) {
+                ATTRIBUTE_NAMES.add(attribute.getName());
+                ATTRIBUTE_FQ_NAMES.add(attribute.getFqName());
```

Comment:
Nitpick: I would rather see these constants to be really constant (also, that would be a bit better from the memory point of view, I think). Something like:

```
List<String> names = new ArrayList<>();
List<String> fqNames = new ArrayList<>();
for (Attributes attribute : Attributes.values()) {
    names.add(attribute.getName());
    fqNames.add(attribute.getFqName());
}
ATTRIBUTE_NAMES = Set.of(names);
ATTRIBUTE_FQ_NAMES = Set.of(fqNames);
```

But this is really a nitpick, feel free to ignore it completely, of course :)

Reply:
Is this OK?(Instead, add attribute names via static methods)

```diff
--- a/php/php.editor/src/org/netbeans/modules/php/editor/PredefinedSymbols.java
+++ b/php/php.editor/src/org/netbeans/modules/php/editor/PredefinedSymbols.java
@@ -107,8 +107,8 @@
                 "__unserialize", // NOI18N PHP 7.4
             })));
 
-    public static final Set<String> ATTRIBUTE_NAMES = Collections.unmodifiableSet(Attributes.ATTRIBUTE_NAMES);
-    public static final Set<String> ATTRIBUTE_FQ_NAMES = Collections.unmodifiableSet(Attributes.ATTRIBUTE_FQ_NAMES);
+    public static final Set<String> ATTRIBUTE_NAMES = Collections.unmodifiableSet(Attributes.getAttributeNames());
+    public static final Set<String> ATTRIBUTE_FQ_NAMES = Collections.unmodifiableSet(Attributes.getAttributeFQNames());
 
     public static enum VariableKind {
         STANDARD,
@@ -129,16 +129,7 @@
         private final String name;
         private final String fqName;
         private final String asAttributeExpression;
-        private static final Set<String> ATTRIBUTE_NAMES = new HashSet<>();
-        private static final Set<String> ATTRIBUTE_FQ_NAMES = new HashSet<>();
 
-        static {
-            for (Attributes attribute : Attributes.values()) {
-                ATTRIBUTE_NAMES.add(attribute.getName());
-                ATTRIBUTE_FQ_NAMES.add(attribute.getFqName());
-            }
-        }
-
         private Attributes(String name) {
             this.name = name;
             this.fqName = "\\" + name; // NOI18N
@@ -156,8 +147,24 @@
         public String asAttributeExpression() {
             return asAttributeExpression;
         }
+
+        private static Set<String> getAttributeNames() {
+            Set<String> names = new HashSet<>();
+            for (Attributes attribute : Attributes.values()) {
+                names.add(attribute.getName());
     }
+            return names;
+        }
 
+        private static Set<String> getAttributeFQNames() {
+            Set<String> names = new HashSet<>();
+            for (Attributes attribute : Attributes.values()) {
+                names.add(attribute.getFqName());
+            }
+            return names;
+        }
+    }
+
     private static String docURLBase;
 
     private PredefinedSymbols() {

```

Reply:
It is the same because it uses HashSet (which is backed by HashMap). But forget it, it is totally fine :)

Reply:
OK. I'll leave it as it is :) I'll change the source level when we use `Set.of()` next time. (Maybe, NB should be required Java 11 at least although I could be wrong.)